src/Zynga/Framework/IO/Web/V1/Curl/CurlRequest.hh

```diff
@@ -0,0 +1,29 @@
+<?hh // strict
+
+namespace Zynga\Framework\IO\Web\V1\Curl;
+use Zynga\Framework\Type\V1\UrlBox;
+
+class CurlRequest implements CurlInterface {
+  private mixed $curlHandle;
```

Comment:
Hacklang provides 'resource' intrinsic to be able to keep track of this as a resource type. I think Jarred had problems with it though (I can't recall if it was when he kept a reference to it as a resource or if it was when it was in use.) Can we change this to see if that will work?
https://docs.hhvm.com/hack/types/type-system

Reply:
Oh cool i didnt know there was a resource type! i can use that and add some mixed checks